pkg/addons/gcpauth/enable.go

```diff
@@ -0,0 +1,105 @@
+/*
+Copyright 2020 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package gcpauth
+
+import (
+	"context"
+	"os"
+	"os/exec"
+	"strconv"
+
+	"github.com/pkg/errors"
+	"golang.org/x/oauth2/google"
+	"k8s.io/minikube/pkg/minikube/assets"
+	"k8s.io/minikube/pkg/minikube/config"
+	"k8s.io/minikube/pkg/minikube/mustload"
+)
+
+const (
+	credentialsPath = "/var/lib/minikube/google_application_credentials.json"
+	projectPath     = "/var/lib/minikube/google_cloud_project"
+)
+
+// EnableOrDisable enables or disables the metadata addon depending on the val parameter
+func EnableOrDisable(cfg *config.ClusterConfig, name string, val string) error {
+	enable, err := strconv.ParseBool(val)
+	if err != nil {
+		return errors.Wrapf(err, "parsing bool: %s", name)
+	}
+	if enable {
+		return enableAddon(cfg)
+	}
+	return disableAddon(cfg)
+
+}
+
+func enableAddon(cfg *config.ClusterConfig) error {
+	// Grab command runner from running cluster
+	cc := mustload.Running(cfg.Name)
+	r := cc.CP.Runner
+
+	// Grab credentials from where GCP would normally look
+	ctx := context.Background()
+	creds, err := google.FindDefaultCredentials(ctx)
+	if err != nil {
+		return err
+	}
+
+	f := assets.NewMemoryAssetTarget(creds.JSON, credentialsPath, "0444")
+
+	err = r.Copy(f)
+	if err != nil {
+		return err
+	}
+
+	// First check if the project env var is explicitly set
+	projectEnv := os.Getenv("GOOGLE_CLOUD_PROJECT")
+	if projectEnv != "" {
+		f := assets.NewMemoryAssetTarget([]byte(projectEnv), projectPath, "0444")
+		return r.Copy(f)
+	}
+
+	// We're currently assuming gcloud is installed and in the user's path
+	project, err := exec.Command("gcloud", "config", "get-value", "project").Output()
+	if err == nil && len(project) > 0 {
+		f := assets.NewMemoryAssetTarget(project, projectPath, "0444")
+		return r.Copy(f)
+	}
+
+	return nil
+}
+
+func disableAddon(cfg *config.ClusterConfig) error {
+	// Grab command runner from running cluster
+	cc := mustload.Running(cfg.Name)
+	r := cc.CP.Runner
+
+	// Clean up the files generated when enabling the addon
+	creds := assets.NewMemoryAssetTarget([]byte{}, credentialsPath, "0444")
+	err := r.Remove(creds)
+	if err != nil {
+		return err
+	}
+
+	project := assets.NewMemoryAssetTarget([]byte{}, projectPath, "0444")
+	err = r.Remove(project)
+	if err != nil {
+		return err
+	}
+
```

Comment:
after enalbing this addon, there should be an out.T

that tells the user what risks they might have

"After enabliing this addon your GCP creds will be mounted to all the created pods, if you dno't want a pod to have the secret, add this label"

Reply:
done

enabling the addon now looks like

```
mk addons enable gcp-auth
🔎  Verifying gcp-auth addon...
📌  Your GCP credentials will now be mounted into every pod created in the minikube cluster.
📌  If you don't want credential mounted into a specific pod, add a label with the `gcp-auth-skip-secret` key to your pod configuration.
🌟  The 'gcp-auth' addon is enabled
```